dash/dash-renderer/src/actions/utils.js

```diff
@@ -37,11 +48,44 @@ export const crawlLayout = (
         // children array
         object.forEach((child, i) => {
             if (extraPath) {
-                crawlLayout(
-                    rpath(extraPath, child),
-                    func,
-                    concat(currentPath, concat([i], extraPath))
-                );
+                const objOf = findIndex(p => includes('{}', p), extraPath);
+                if (objOf !== -1) {
+                    const front = slice(0, objOf, extraPath);
+                    const back = slice(objOf, extraPath.length, extraPath);
+                    if (front.length) {
+                        crawlLayout(
+                            rpath(front, child),
+                            func,
+                            concat(currentPath, concat([i], front)),
+                            back
+                        );
+                    } else {
+                        const backPath = back
+                            .map(p => p.replace('{}', ''))
+                            .filter(e => e);
+                        let childObj,
+                            childPath = concat([i], backPath);
+                        if (backPath.length) {
+                            childObj = rpath(backPath, child);
+                        } else {
+                            childObj = child;
+                        }
+                        Object.keys(childObj).forEach(key => {
+                            const value = childObj[key];
+                            crawlLayout(
+                                value,
+                                func,
+                                concat(currentPath, childPath.concat([key]))
+                            );
+                        });
```

Comment:
Not a big deal, just a little cleaner. Same at [line 143](https://github.com/plotly/dash/pull/2207/files#diff-7547342484ca8c69ce2849cd1e8f7e8819b751cbd91e3f315368ca6f397b896dR143)
```suggestion
                        for (const key in childObj) {
                            const value = childObj[key];
                            crawlLayout(
                                value,
                                func,
                                concat(currentPath, childPath.concat([key]))
                            );
                        }
```